File: leetcode2025/2025_02/leetcode2025-02-03.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        n = len(s)
        def myCheck(op):
            i, j = 0, n-1
            tmp = 0
            while i<j:
                if s[i]!=s[j]:
                    if tmp==1: return False
                    tmp = 1
                    if op==0:
                        i += 1
                    else:
                        j -= 1
                else:
                    i += 1
                    j -= 1
            return True
        return myCheck(0) | myCheck(1)

if __name__ == "__main__":
    sl = Solution()
    s = "abc"
    print(sl.validPalindrome(s))

File: leetcode2025/2025_02/leetcode2025-02-03.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
private:
    int n;
    bool myCheck(int op, string s){
        int i=0, j=n-1;
        int tmp = 0;
        while (i<j){
            if (s[i]!=s[j]){
                if (tmp==1) return false;
                tmp = 1;
                if (op==0){
                    i++;
                }
                else{
                    j--;
                }
            }
            else{
                i++;
                j--;
            }
        }
        return true;
    }
public:
    bool validPalindrome(string s) {
        n = s.size();
        return myCheck(0, s) | myCheck(1, s);
    }
};

int main(){
    Solution sl;
    string s = "abc";
    cout << sl.validPalindrome(s) << endl;
}
